### DISCUSSION TITLE
[SOLVED] [FR]: Custom UI color themes? edit existing ones?

### DISCUSSION THREAD

**Discussion by Owyn** [martinrotter/rssguard#635] [Q&A] [answered]
### Brief description of the feature request

I couldn't find themes in the apps folder anywhere to edit it (without recompiling everything) , it would be nice if we could do that - people would be interested in making custom color themes

**Comment by martinrotter** [comment, marked as answer]
You already can do custom themes! First read "Skins" section [here](https://github.com/martinrotter/rssguard/blob/master/resources/docs/Documentation.md#skin). (I plan to update this part of docs).

Basically, you create "skins" subfolder in your user data folder and then copy some existing skin for example ["plain"](https://github.com/martinrotter/rssguard/tree/master/resources/skins/plain) into the folder, so you will have "......\data4\skin\my-plain\" path. RSS Guard on restart will detect the skin and you can activate it. Files inside the "plain" are documented, so you can have some idea how to use them.

**Reply by Owyn** [reply]
I didn't want to create a brand new skin tho... I just wanted to edit few things in the existing one

Can you maybe share existing themes in that format which can be used in the skins subfolder?

or will https://github.com/martinrotter/rssguard/tree/master/resources/skins/nudus-dark work for that?

**Reply by martinrotter** [reply]
Then you simply copy the existing skin you want to tweak and work on that. :)

Check here: https://github.com/martinrotter/rssguard/tree/master/resources/skins

From that folder, simply copy skin you want, specifically: plain, nudus-light, nudus-dark can be simply copied and worked with but I recommend to try with "plain". Also, file https://github.com/martinrotter/rssguard/blob/master/resources/skins/plain/qt_style.qss is quite basic one, if you want some more deep template of QT CSS to start with then you can for example use this: https://github.com/QuasarApp/QStyleSheet/blob/master/DarkStyle/darkStyle.css (simply copy contents into "qt_style.qss" and it will work.

**Reply by Owyn** [reply]
Hmmm... I copied nudus-dark folder into my data4\skins folder and now app just crashes silently when I open settings -> user interface :-( I didn't even edit anything in it

**Reply by martinrotter** [reply]
ok let me check

**Reply by martinrotter** [reply]
> Hmmm... I copied nudus-dark folder into my data4\skins folder and now app just crashes silently when I open settings -> user interface :-( I didn't even edit anything in it

what app version?

**Reply by Owyn** [reply]
RSS Guard
Version: 4.2.0 (built on Windows/AMD64)
Revision: 69bcff16-nowebengine

**Reply by martinrotter** [reply]
what error does it show in log? can you generate log for me?

**Reply by martinrotter** [reply]
EDIT: Did you rename "nudus-dark" folder to something elses?

**Reply by martinrotter** [reply]
Found a problem, nudus-dark uses special mechanism we call "base skin". It is "derived" skin. If you want to really work from nudus-dark (or nudus-light) you need to ALSO copy to your folder parent skin called "nudus-base".

I would really recommend to start with plain.

**Reply by martinrotter** [reply]
[plain.zip](https://github.com/martinrotter/rssguard/files/8056331/plain.zip)

Here is sample "plain" with tweaked QT CSS i made for you, try it.

**Reply by Owyn** [reply]
> EDIT: Did you rename "nudus-dark" folder to something elses?

yea, I renamed it to `nudus-dark-my`

I've just tried the plain, created a 'plain' folder and put there just metadata file - still crashed

but your plain folder worked

**Reply by martinrotter** [reply]
> I've just tried the plain, created a 'plain' folder and put there just metadata file - still crashed

That won't work, because just "metadata.xml" file is not enough, you need almost all regular files, like from that plain folder i sent to you. You basically can leave some files intact, it depends what exact part of UI you want to tweak etc.

**Comment by martinrotter** [comment]
Also, turing to discussion.

**Comment by Owyn** [comment]
I've actually found what I wanted and don't need to edit the theme anymore...

I just wanted to change the color of **"Interesting stuff"** (whatever that means...)
![image](https://user-images.githubusercontent.com/1309656/153773446-130a26ff-eb47-456f-acdf-27c3cff775af.png)

but I still have absolutely no idea what green "Ok-ish stuff" is supposed to be about
